relay-server/src/actors/processor.rs

```diff
@@ -1995,6 +2031,7 @@ impl EnvelopeProcessorService {
         }
 
         if_processing!({
+            // TODO: should see it is already rate-limited and not use redis if so
```

Comment:
enfoce_quotas needs to remove the transaction event if index quota is exhausted so that serialize_event does not get to use it